controllers/article/LikeController.js

```diff
@@ -0,0 +1,103 @@
+import db from '../../models';
+import ArticleHelper from '../../helpers/ArticleHelper';
+import HttpError from '../../helpers/exceptionHandler/httpError';
+
+const {
+  Like, User, Comment, CommentReaction
+} = db;
+/**
+ *
+ * @description Handles user article's like
+ * @export
+ * @class LikeController
+ */
+export default class LikeController {
+  /**
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @memberof LikeController
+   * @return {json} Returns json object
+   */
+  static async addLike(req, res) {
+    const { slug, status } = req.params;
+    const { userId } = req.authData;
+
+    const article = await ArticleHelper.findArticleBySlug(slug);
+
+    HttpError.throwErrorIfNull(article, 'Article not found');
+
+    const articleId = article.id;
+    const existingLike = await Like.findOne({ where: { $and: [{ articleId }, { userId }] } });
+
+    if (existingLike) {
+      await Like.update({ status }, { where: { $and: [{ articleId }, { userId }] } });
+      return res.status(200)
+        .json({
+          status: 'success',
+          message: 'successfully modified like status',
+        });
+    }
+    Like.create({
```

Comment:
Good job on your implementation on this feature, however, it would be nice to await the response you get from creating like and check if it is successful before returning a 201 status code.

Reply:
Thanks for pointing it out. I will resolve it as soon as possible